drone-webdriver/src/test/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/GitHubSourceAuthenticatedRequestTestCase.java

```diff
@@ -0,0 +1,70 @@
+package org.jboss.arquillian.drone.webdriver.binary.downloading.source;
+
+import io.specto.hoverfly.junit.rule.HoverflyRule;
+import java.util.Base64;
+import java.util.logging.Logger;
+import org.jboss.arquillian.drone.webdriver.utils.GitHubLastUpdateCache;
+import org.jboss.arquillian.drone.webdriver.utils.HttpClient;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.openqa.selenium.remote.DesiredCapabilities;
+
+import static io.specto.hoverfly.junit.core.SimulationSource.classpath;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class GitHubSourceAuthenticatedRequestTestCase {
+
+    private static final Logger log = Logger.getLogger(GitHubSourceAuthenticatedRequestTestCase.class.getName());
+
+    @ClassRule
+    public static HoverflyRule hoverflyRule =
+        HoverflyRule.inSimulationMode(classpath("hoverfly/gh.simulation.mozilla@geckodriver.releases.json"));
+
+    @Rule
+    public final TemporaryFolder folder = new TemporaryFolder();
+
+    @Test
+    public void should_add_authentication_header_to_request() throws Exception {
+        // given
+        HttpClient httpClientSpy = spy(new HttpClient());
+        GitHubLastUpdateCache gitHubLastUpdateCache = new GitHubLastUpdateCache(folder.newFolder());
+
+        DesiredCapabilities capabilities = new DesiredCapabilities();
+        capabilities.setCapability(GitHubSource.GITHUB_USERNAME_PROPERTY, "my-username");
+        capabilities.setCapability(GitHubSource.GITHUB_TOKEN_PROPERTY, "my-token");
+
+        GeckoDriverGitHubSource geckoDriverGitHubSource = new GeckoDriverGitHubSource(httpClientSpy, gitHubLastUpdateCache, capabilities);
+
+        // when
+        geckoDriverGitHubSource.getReleaseForVersion("v0.14.0");
+
+        // then
+        verify(httpClientSpy, times(1)).get(anyString(), argThat(headers -> {
+            if (headers.size() > 0) {
+                String authorization = headers.get("Authorization");
```

Comment:
Maybe for better readability extract to a new method so we know what is the topic of verification here?